SingleRowFunction not resolved with single "." string literal parameter

When I have a single-row function with a single string parameter, the compiler always fails when I try to pass a string literal with a dot in it.

``` java
import com.espertech.esper.common.client.configuration.Configuration;
import com.espertech.esper.compiler.client.CompilerArguments;
import com.espertech.esper.compiler.client.EPCompilerProvider;

public class Main {
    public static void main(String[] args) throws Exception {
        var compiler = EPCompilerProvider.getCompiler();
        var configuration = new Configuration();
        configuration.getCompiler().addPlugInSingleRowFunction("Bar", Main.class.getCanonicalName(), "Bar");
        var arguments = new CompilerArguments(configuration);

        // this works fine
        compiler.compile("create variable Object Foo = Bar(\"test\")", arguments);

        // this doesn't work
        compiler.compile("create variable Object Foo = Bar(\".\")", arguments);
    }

    public static String Bar(String input) {
        return input;
    }
}
```

```
com.espertech.esper.compiler.client.EPCompileException: Failed to validate variable-assignment expression 'Bar(".")': Failed to resolve enumeration method, date-time method or mapped property 'Bar(".")': Failed to resolve '.Bar' to a property, single-row function, aggregation function, script, stream or class name [create variable Object Foo = Bar(".")]
	at com.espertech.esper.compiler.internal.util.CompilerHelperModuleProvider.compileToBytes(CompilerHelperModuleProvider.java:150)
	at com.espertech.esper.compiler.internal.util.CompilerHelperModuleProvider.compile(CompilerHelperModuleProvider.java:82)
	at com.espertech.esper.compiler.internal.util.EPCompilerImpl.compile(EPCompilerImpl.java:89)
	at Main.main(Main.java:19)
Caused by: com.espertech.esper.compiler.client.EPCompileExceptionItem: Failed to validate variable-assignment expression 'Bar(".")': Failed to resolve enumeration method, date-time method or mapped property 'Bar(".")': Failed to resolve '.Bar' to a property, single-row function, aggregation function, script, stream or class name
	at com.espertech.esper.compiler.internal.util.CompilerHelperModuleProvider.compileToBytes(CompilerHelperModuleProvider.java:125)
	... 3 more
Caused by: com.espertech.esper.common.internal.compile.stage2.StatementSpecCompileException: Failed to validate variable-assignment expression 'Bar(".")': Failed to resolve enumeration method, date-time method or mapped property 'Bar(".")': Failed to resolve '.Bar' to a property, single-row function, aggregation function, script, stream or class name
	at com.espertech.esper.compiler.internal.util.CompilerHelperStatementProvider.compileItem(CompilerHelperStatementProvider.java:278)
	at com.espertech.esper.compiler.internal.util.CompilerHelperModuleProvider.compileToBytes(CompilerHelperModuleProvider.java:110)
	... 3 more
Caused by: com.espertech.esper.common.internal.epl.expression.core.ExprValidationException: Failed to validate variable-assignment expression 'Bar(".")': Failed to resolve enumeration method, date-time method or mapped property 'Bar(".")': Failed to resolve '.Bar' to a property, single-row function, aggregation function, script, stream or class name
	at com.espertech.esper.common.internal.epl.expression.core.ExprNodeUtilityValidate.getValidatedSubtree(ExprNodeUtilityValidate.java:137)
	at com.espertech.esper.common.internal.context.aifactory.createvariable.StmtForgeMethodCreateVariable.make(StmtForgeMethodCreateVariable.java:77)
	at com.espertech.esper.compiler.internal.util.CompilerHelperStatementProvider.compileItem(CompilerHelperStatementProvider.java:214)
	... 4 more
Caused by: com.espertech.esper.common.internal.epl.expression.core.ExprValidationException: Failed to resolve enumeration method, date-time method or mapped property 'Bar(".")': Failed to resolve '.Bar' to a property, single-row function, aggregation function, script, stream or class name
	at com.espertech.esper.common.internal.epl.expression.core.ExprNodeUtilityValidate.resolveStaticMethodOrField(ExprNodeUtilityValidate.java:355)
	at com.espertech.esper.common.internal.epl.expression.core.ExprNodeUtilityValidate.getValidatedSubtreeInternal(ExprNodeUtilityValidate.java:254)
	at com.espertech.esper.common.internal.epl.expression.core.ExprNodeUtilityValidate.getValidatedSubtree(ExprNodeUtilityValidate.java:118)
	... 6 more

Process finished with exit code 0
```

--- a/common/src/main/java/com/espertech/esper/common/internal/event/propertyparser/PropertyParserNoDep.java
+++ b/common/src/main/java/com/espertech/esper/common/internal/event/propertyparser/PropertyParserNoDep.java
@@ -56,90 +56,57 @@ public class PropertyParserNoDep {
      * @return descriptor object
      */
     public static MappedPropertyParseResult parseMappedProperty(String property) {
-        // get argument
-        int indexFirstDoubleQuote = property.indexOf("\"");
-        int indexFirstSingleQuote = property.indexOf("'");
-        int startArg;
-        if ((indexFirstSingleQuote == -1) && (indexFirstDoubleQuote == -1)) {
+        // split the class and method from the parentheses and argument
+        int indexOpenParen = property.indexOf("(");
+        if (indexOpenParen == -1) {
             return null;
         }
-        if ((indexFirstSingleQuote != -1) && (indexFirstDoubleQuote != -1)) {
-            if (indexFirstSingleQuote < indexFirstDoubleQuote) {
-                startArg = indexFirstSingleQuote;
-            } else {
-                startArg = indexFirstDoubleQuote;
-            }
+        String classAndMethod = property.substring(0, indexOpenParen);
+        String parensAndArg = property.substring(indexOpenParen);
+        if (classAndMethod.length() == 0 || parensAndArg.length() == 0) {
+            return null;
+        }
+        // find the first quote
+        int startArg;
+        int indexFirstDoubleQuote = parensAndArg.indexOf("\"");
+        int indexFirstSingleQuote = parensAndArg.indexOf("'");
+        if (indexFirstSingleQuote != -1 && indexFirstDoubleQuote != -1) {
+            startArg = Math.min(indexFirstDoubleQuote, indexFirstSingleQuote);
         } else if (indexFirstSingleQuote != -1) {
             startArg = indexFirstSingleQuote;
-        } else {
+        } else if (indexFirstDoubleQuote != -1) {
             startArg = indexFirstDoubleQuote;
-        }
-
-        int indexLastDoubleQuote = property.lastIndexOf("\"");
-        int indexLastSingleQuote = property.lastIndexOf("'");
-        int endArg;
-        if ((indexLastSingleQuote == -1) && (indexLastDoubleQuote == -1)) {
-            return null;
-        }
-        if ((indexLastSingleQuote != -1) && (indexLastDoubleQuote != -1)) {
-            if (indexLastSingleQuote > indexLastDoubleQuote) {
-                endArg = indexLastSingleQuote;
-            } else {
-                endArg = indexLastDoubleQuote;
-            }
-        } else if (indexLastSingleQuote != -1) {
-            if (indexLastSingleQuote == indexFirstSingleQuote) {
-                return null;
-            }
-            endArg = indexLastSingleQuote;
         } else {
-            if (indexLastDoubleQuote == indexFirstDoubleQuote) {
-                return null;
-            }
+            return null;
+        }
+        // find the last quote
+        int endArg;
+        int indexLastDoubleQuote = parensAndArg.lastIndexOf("\"");
+        int indexLastSingleQuote = parensAndArg.lastIndexOf("'");
+        if (indexLastSingleQuote != -1 && indexLastDoubleQuote != -1) {
+            endArg = Math.max(indexLastDoubleQuote, indexLastSingleQuote);
+        } else if (indexLastSingleQuote != -1) {
+            endArg = indexLastSingleQuote;
+        } else if (indexLastDoubleQuote != -1) {
             endArg = indexLastDoubleQuote;
-        }
-        String argument = property.substring(startArg + 1, endArg);
-
-        // get method
-        String[] splitDots = property.split("[\\.]");
-        if (splitDots.length == 0) {
+        } else {
             return null;
         }
-
-        // find which element represents the method, its the element with the parenthesis
-        int indexMethod = -1;
-        for (int i = 0; i < splitDots.length; i++) {
-            if (splitDots[i].contains("(")) {
-                indexMethod = i;
-                break;
-            }
-        }
-        if (indexMethod == -1) {
+        if (startArg == endArg) {
             return null;
         }
-
-        String method = splitDots[indexMethod];
-        int indexParan = method.indexOf("(");
-        method = method.substring(0, indexParan);
+        String argument = parensAndArg.substring(startArg + 1, endArg);
+        // split the class from the method
+        int indexLastDot = classAndMethod.lastIndexOf(".");
+        if (indexLastDot == -1) {
+            // no class name
+            return new MappedPropertyParseResult(null, classAndMethod, argument);
+        }
+        String method = classAndMethod.substring(indexLastDot + 1);
         if (method.length() == 0) {
             return null;
         }
-
-        if (splitDots.length == 1) {
-            // no class name
-            return new MappedPropertyParseResult(null, method, argument);
-        }
-
-
-        // get class
-        StringBuilder clazz = new StringBuilder();
-        for (int i = 0; i < indexMethod; i++) {
-            if (i > 0) {
-                clazz.append('.');
-            }
-            clazz.append(splitDots[i]);
-        }
-
-        return new MappedPropertyParseResult(clazz.toString(), method, argument);
+        String clazz = classAndMethod.substring(0, indexLastDot);
+        return new MappedPropertyParseResult(clazz, method, argument);
     }
 }

--- a/common/src/test/java/com/espertech/esper/common/internal/event/propertyparser/TestPropertyParserNoDep.java
+++ b/common/src/test/java/com/espertech/esper/common/internal/event/propertyparser/TestPropertyParserNoDep.java
@@ -111,6 +111,11 @@ public class TestPropertyParserNoDep extends TestCase {
         assertEquals("f", result.getMethodName());
         assertEquals("a", result.getArgString());
 
+        result = parseMappedProperty("f('.')");
+        assertEquals(null, result.getClassName());
+        assertEquals("f",  result.getMethodName());
+        assertEquals(".", result.getArgString());
+
         assertNull(parseMappedProperty("('a')"));
         assertNull(parseMappedProperty(""));
     }

